scripts/update/main.go

```diff
@@ -61,12 +74,94 @@ func GetBazelDeps(repo RepositoryDepsURL) []Repository {
 
 	// TODO: Replace with a starlark interpreter that can be used for any project.
 	for _, match := range re.FindAllString(fc, -1) {
-		repos = append(repos, Repository{match, ""})
+		repos = append(repos, Repository{strings.TrimSuffix(match, ".git"), ""})
 	}
 
 	return repos
 }
 
+// GetGoDeps returns go repo dependencies.
+func GetGoDeps(repo RepositoryDepsURL) []Repository {
+	client := github.NewClient(nil)
+	ctx := context.Background()
+	repos := []Repository{}
+	fo, _, _, err := client.Repositories.GetContents(ctx, repo.Owner, repo.Repo, repo.File, nil)
+	if err != nil {
+		// If we can't get content, gracefully fail but alert.
+		log.Panicf("Failed to get repository content %s", err)
+		return repos
+	}
+
+	fc, err := fo.GetContent()
+	if err != nil {
+		// If we can't get content, gracefully fail, but alert.
+		log.Panicf("Failed to get repository content %s", err)
+		return repos
+	}
+
+	tmpFile, err := ioutil.TempFile(os.TempDir(), "")
+	if err != nil {
+		log.Fatal("Cannot create temporary file", err)
+	}
+
+	// Remember to clean up the file afterwards
+	defer os.Remove(tmpFile.Name())
+
+	// Example writing to the file
+	if _, err = tmpFile.Write([]byte(fc)); err != nil {
+		log.Panicf("Failed to write to temporary file %s", err)
+	}
+
+	//nolint
+	cmd := exec.Command("go", "mod", "edit", "--json", tmpFile.Name())
+
+	var out bytes.Buffer
+	cmd.Stdout = &out
+
+	err = cmd.Run()
+	if err != nil {
+		log.Panic(err)
+	}
+
+	mods := &gomod{}
+	if err := json.Unmarshal(out.Bytes(), mods); err != nil {
+		log.Panic(err)
+	}
+	// Close the file
+	if err := tmpFile.Close(); err != nil {
+		log.Panic(err)
+	}
+
+	for _, match := range mods.Require {
```

Comment:
I think this is actually a common misconception with go.mod files - they do not contain the full set of dependencies required to build an application. For that, you need to use `go list -m all`.

Reply:
I am guessing `go list -m all` gives transitive dependency. Do we need the transitive dependencies?

Reply:
I don't think it's that simple but can't explain it perfectly myself. My understanding is that go.mod only contains dependencies that are not the "natural" versions that would be selected by MVS.

Reply:
So you think we should change the implementation to use the go list -m all instead of go.mod?

Reply:
Yeah, I understand "go list -m all", but I don't understand go.mod :)

It's possible that go.mod is all we need if we only care about direct dependencies, but from a security standpoint transitive dependencies are just as important.

Reply:
I understand. I am going to try and see if we can go list -m all with just go.mod and go.sum.

Reply:
OK, my doubt was correct, can't get the `go list -m all` with just `go.sum` and `go.mod`. We would have to download the entire repository.

Reply:
I am going to change the implementation to download the repository and then run `go list -m all`

Reply:
Took care of it. Thanks